lib/Sema/CSApply.cpp

```diff
@@ -7785,6 +7785,21 @@ bool ConstraintSystem::applySolutionFix(Expr *expr,
   }
     
   switch (fix.first.getKind()) {
+  case FixKind::OptionalChaining: {
+    if (!solution.DisjunctionChoices.count(locator) ||
```

Comment:
Examining disjuncts is a frighteningly brittle way to propagate this through the constraint system.  I wonder if you couldn't implement this with another `FixKind`.

Reply:
It's the way that implicit unwrapping or not with IUOs gets propagated. I know IUOs isn't exactly the best part of the implementation to emulate, but it is existing practice... If you have another suggestion I'd be glad to try it out. Another ```FixKind``` wouldn't work by itself because there is no way to know at the time the fix is added which of the two possibilities will type check.

Reply:
>  it is existing practice

It is existing practice in the context of the implementation of IUOs because there we put up a front to maintain the illusion that IUOs aren't a modality of `T?`.  This uses disjunctions as a pieces of state and I suspect we have better ways of handling that.